src/main/java/tc/oc/pgm/rotation/RotationManager.java

```diff
@@ -0,0 +1,220 @@
+    rotations.forEach(
+        rotation -> {
+          if (rotation.isEnabled()) {
+            playerCounts[count.get()] = rotation.getPlayers();
+            count.getAndIncrement();
+          }
+        });
+
+    int distance = Math.abs(playerCounts[0] - activePlayers);
+    int accuratePlayerCountIndex = 0;
+
+    for (int i = 1; i < playerCounts.length; i++) {
+      int idistance = Math.abs(playerCounts[i] - activePlayers);
+      if (idistance < distance) {
+        accuratePlayerCountIndex = i;
+        distance = idistance;
+      }
+    }
+
+    updateRotation(playerCounts, accuratePlayerCountIndex);
+  }
+
+  private int getActivePlayers() {
+    Collection<? extends Player> onlinePlayers = Bukkit.getOnlinePlayers();
+    if (onlinePlayers.isEmpty()) return 0;
+
+    Match match = matchManager.getMatch((Player) onlinePlayers.toArray()[0]);
+
+    assert match != null;
+    return onlinePlayers.size() - match.getObservers().size() / 2;
+  }
+
+  private void updateRotation(final int[] playerCounts, final int accuratePlayerCountIndex) {
+    rotations.forEach(
+        rotation -> {
+          if (rotation.getPlayers() == playerCounts[accuratePlayerCountIndex]
+              && activeRotation.getPlayers() != playerCounts[accuratePlayerCountIndex]) {
+            setActiveRotation(rotation);
+            Bukkit.broadcastMessage(
+                ChatColor.WHITE
+                    + "["
+                    + ChatColor.GOLD
+                    + "Rotations"
+                    + ChatColor.WHITE
+                    + "] "
+                    + ChatColor.GREEN
+                    + AllTranslations.get()
+                        .translate(
+                            "rotations.rotationChange",
+                            Bukkit.getConsoleSender(),
+                            (ChatColor.AQUA + rotation.getName() + ChatColor.GREEN)));
+          }
+        });
+  }
+
+  public Rotation getRotationByName(String name) {
+    AtomicReference<Rotation> matchByName = new AtomicReference<>();
+
+    getRotations()
+        .forEach(
+            rotation -> {
+              if (rotation.getName().equals(name)) {
+                matchByName.set(rotation);
+              }
+            });
+
+    return matchByName.get();
+  }
+
+  public PGMMap getInitialMap() {
+    return activeRotation.getMapInPosition(activeRotation.getPosition());
+  }
+
+  @Override
+  public PGMMap popNextMap() {
+    if (overriderMap == null) {
+      saveCurrentPosition();
+      return activeRotation.popNextMap();
+    } else {
+      PGMMap overrider = overriderMap;
+      overriderMap = null;
+      return overrider;
+    }
+  }
+
+  @Override
+  public PGMMap getNextMap() {
+    if (activeRotation != null) {
+      if (overriderMap == null) {
+        return activeRotation.getNextMap();
+      } else return overriderMap;
+    } else return popFallbackMap();
+  }
+
+  @Override
+  public PGMMap popFallbackMap() {
```

Comment:
Isn't this just: getRandomMap?

Reply:
I was encouraged not to use getRandomMap anymore... I also throught it would be. @Pablete1234 ):

Reply:
This method should probably not exist at all.
If the rotation failed to load, pgm should just load a RandomMapOrder (pick random map in the map pool) or a DefaultMapOrder (single map) for pgm to work

Reply:
I have created a type of `PGMMapOrder` called `RandomPGMMapOrder`, hopefully that will do for this. https://github.com/Electroid/PGM/pull/114/commits/fe83e3a18ec9419e8184475a0ccd8c4fb5a5e93a